lblod/app-digitaal-loket
# Adding db cleanup migrations

# Description

DL-5601

This PR adds new migrations for the db cleanup service. See changes in https://github.com/lblod/db-cleanup-service/pull/1

After testing locally jobs are running and resources are successfully deleted but there is cases where a job has a lot of resources, thinking about this culprit <http://lblod.data.gift/id/jobs/concept/JobOperation/downloadUrlWarning>
```ttl
<http://data.lblod.info/id/cleanup-job/3f623084-416f-44ef-90fa-7c7fa3d4efad> a cleanup:Job ;
    mu:uuid "3f623084-416f-44ef-90fa-7c7fa3d4efad" ;
    dcterms:title "clean up more than a month old jobs and tasks from harvesting graph" ;
    cleanup:selectPattern """
              GRAPH <http://mu.semte.ch/graphs/harvesting> {
                VALUES ?operation {
                  <http://lblod.data.gift/id/jobs/concept/JobOperation/berichtencentrumWarning>
                  <http://lblod.data.gift/id/jobs/concept/JobOperation/downloadUrlWarning>
                }
                ?resource a <http://vocab.deri.ie/cogs#Job> ;
                  <http://redpencil.data.gift/vocabularies/tasks/operation> ?operation;
                  <http://purl.org/dc/terms/modified> ?modified .
                FILTER (?modified < (NOW() - "P1M"^^xsd:duration))
                ?task <http://purl.org/dc/terms/isPartOf> ?resource.
                ?resource ?p ?o.
                ?task ?taskP ?taskO.
                OPTIONAL {
                  ?task <http://redpencil.data.gift/vocabularies/tasks/resultsContainer> ?resultsContainer .
                  ?resultsContainer ?presultsContainer ?oresultsContainer .
                }
                OPTIONAL {
                  ?task <http://redpencil.data.gift/vocabularies/tasks/inputContainer> ?inputContainer .
                  ?inputContainer ?pinputContainer ?oinputContainer .
                }
              }
              """ ;
    cleanup:deletePattern """
             GRAPH <http://mu.semte.ch/graphs/harvesting> {
               ?resource ?p ?o.
               ?task ?taskP ?taskO.
               ?resultsContainer ?presultsContainer ?oresultsContainer .
               ?inputContainer ?pinputContainer ?oinputContainer .
             }
              """ ;
    cleanup:cronPattern "*/45 * * * *" . 
``` 
As an indication, there are more than a million resources to be deleted locally, which may take some time until all the operations are completed, so they should be spaced out over time with other resource-consuming operations.

# Type of change


# Related services

- N/A

# How to test

:warning: **Important:  Make sure to make a backup of your data to test this feature**

1. Link PR volume or use image in loket
2. Add cleanup migrations through Loket PR (link pr here)
3. Let the service run it should delete resources at configured cronPattern
 
# What to check

- Decide which cronPattern might fit for each jobs

# Links to other PR's

- https://github.com/lblod/db-cleanup-service/pull/1

# Notes

- N/A
